src/pkg/util/env/clean.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package env
+
+import (
+	"strings"
+
+	"github.com/opencontainers/runtime-tools/generate"
+	"github.com/singularityware/singularity/src/pkg/sylog"
+)
+
+// CleanEnv cleans environment variables before running the container
+func CleanEnv(g *generate.Generator, IsCleanEnv bool, environment []string) {
+
+	for _, env := range environment {
+		e := strings.SplitN(env, "=", 2)
+		if len(e) != 2 {
+			sylog.Verbosef("Can't process environment variable %s", env)
+			continue
+		}
+
+		// Transpose environment
+		if strings.HasPrefix(e[0], "SINGULARITYENV_") {
+			e[0] = strings.TrimPrefix(e[0], "SINGULARITYENV_")
+		} else if IsCleanEnv && (e[0] != "HOME" &&
```

Comment:
This seems like we're doing things in a very naive way here. First of all, we're not guaranteed that these variables are set yet. For instance in 2.x, we always force the env var `LANG=C` to be set (however, this was to fix some bugs and I *don't* think that needs to happen in 3.0)

Reply:
I spoke with Greg about the `LANG=C` thing and I think he just set it b/c it is critical for many apps (like Perl) and he was tired of getting issues raised about things being broken because of it.  IMHO we should leave it because we have already gone back and forth about it once.  It's a judgement call, a judgement has been made, and now it's important to just be consistent.

Reply:
so... final call : 
 - force `LANG=C`?
 - or just pass the `LANG` env value from the host?

Reply:
Always set `LANG=C` I guess

Reply:
From 2.6 
```
$ singularity exec ~/tmp/sing/centos7.img env | grep ^LANG
LANG=en_US
$ singularity exec --cleanenv ~/tmp/sing/centos7.img env | grep ^LANG 
LANG=C
```
so force `LANG=C` is just when using `--cleanenv` flag